### PR TITLE
GSLUX-265: import style

### PR DESCRIPTION
**Pull request by llienher**
https://jira.camptocamp.com/browse/GSLUX-265

### CODE REVIEW

**Comment by jaykayone** [comment]
We currently have an issue with the staging tileserver, so I made a commit to change the URLs.
Please make sure we don't get any conflicts here

File: geoportal/geoportailv3_geoportal/static-ngeo/js/backgroundlayer/BackgroundlayerController.js [hunk 1]
```diff
@@ -59,8 +60,53 @@ class Controller {
      * @type {import('ol/Map').default}
      */
     this.map;
+
+    /**
+     * @type {boolean}
+     */
+    this.activeMvt;
+
+    this.setCustomStyle = () => {
```
Comment (gberaudo) [inline geoportal/geoportailv3_geoportal/static-ngeo/js/backgroundlayer/BackgroundlayerController.js]:
```suggestion
    this.setCustomStyle = (event) => {
```

File: geoportal/geoportailv3_geoportal/static-ngeo/js/backgroundlayer/BackgroundlayerController.js [hunk 2]
```diff
@@ -59,8 +60,53 @@ class Controller {
...
+      // If the file is not a JSON, stop here
+      if (file.type !== 'application/json') {
+        return;
+      }
+    
+      // Read the json file, save it to local storage and load it to the map
+      this.readFile_(file, (e) => {
+        const result = e.target.result;
+        this.appThemes_.setCustomVectorTileStyle(this.bgLayer, result);
+        this.hasCustomStyle();
```
Comment (gberaudo) [inline geoportal/geoportailv3_geoportal/static-ngeo/js/backgroundlayer/BackgroundlayerController.js]:
```suggestion
```

File: geoportal/geoportailv3_geoportal/static-ngeo/js/backgroundlayer/BackgroundlayerController.js [hunk 3]
```diff
@@ -59,8 +60,53 @@ class Controller {
...
+      });
+    
+      // Reset form value
+      event.target.value = null;
+      file = null;
+    };
+
+    this.clearCustomStyle = () => {
+      this.appThemes_.setCustomVectorTileStyle(this.bgLayer, undefined);
+      this.hasCustomStyle();
```
Comment (gberaudo) [inline geoportal/geoportailv3_geoportal/static-ngeo/js/backgroundlayer/BackgroundlayerController.js]:
```suggestion
```

File: geoportal/geoportailv3_geoportal/static-ngeo/js/backgroundlayer/BackgroundlayerController.js [hunk 4]
```diff
@@ -28,12 +28,13 @@ import {listen} from 'ol/events.js';
...
 /**
  * @constructor
+  * @param {angular.Scope} $scope Scope.
  * @param {ngeo.map.BackgroundLayerMgr} ngeoBackgroundLayerMgr Background layer manager.
  * @param {import('../Themes').default} appThemes The app themes.
  * @ngInject
  */
 class Controller {
-  constructor(ngeoBackgroundLayerMgr, appThemes) {
+  constructor($scope, ngeoBackgroundLayerMgr, appThemes) {
```
Comment (gberaudo) [inline geoportal/geoportailv3_geoportal/static-ngeo/js/backgroundlayer/BackgroundlayerController.js]: Do you still need the $scope?

File: geoportal/geoportailv3_geoportal/static-ngeo/js/backgroundlayer/backgroundlayer.html [hunk 5]
```diff
@@ -11,3 +11,14 @@
...
+<div class="mvt-style-loader" ng-if="ctrl.activeMvt">
+  <label for="uploadMvtStyle" class="btn">Changer style vectorielle</label>
+  <input
+    class="btn btn-default"
+    type="file"
+    name="uploadMvtStyle"
+    id="uploadMvtStyle"
+    custom-on-change="ctrl.setCustomStyle">
+  <a class="btn btn-default" id="clearMvtStyle" ng-click="ctrl.clearCustomStyle()" ng-if="ctrl.hasCustomStyle()">Supprimer</a>
+</div>
```
Comment (gberaudo) [inline geoportal/geoportailv3_geoportal/static-ngeo/js/backgroundlayer/backgroundlayer.html]: These texts should be translatable, right?

Comment (llienher) [reply geoportal/geoportailv3_geoportal/static-ngeo/js/backgroundlayer/backgroundlayer.html]: Yes, I did the change.

**Pull request merged**